addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeAggregateDataPolling.java

```diff
@@ -0,0 +1,88 @@
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdatePublicApi;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.openhab.binding.solaredge.internal.model.AggregatePeriod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Polling worker class. This is responsible for periodic polling of sensor data.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+public class SolarEdgeAggregateDataPolling implements Runnable {
+    /**
+     * Logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Handler for delegation to callbacks.
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * Constructor.
+     *
+     * @param handler handler which handles results of polling
+     */
+    public SolarEdgeAggregateDataPolling(SolarEdgeHandler handler) {
+        this.handler = handler;
+    }
+
+    /**
+     * Poll the SolarEdge Webservice one time per call.
+     */
+    @Override
+    public void run() {
+        if (handler.getWebInterface() != null) {
+            // if no meter is present all data will be fetched by the 'LiveDataUpdateMeterless'
+            if (handler.getConfiguration().isMeterInstalled()) {
+                logger.debug("polling SolarEdge aggregate data {}", handler.getConfiguration());
+
+                List<SolarEdgeCommand> commands = new ArrayList<>();
+
+                if (handler.getConfiguration().isUsePrivateApi()) {
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.DAY));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.WEEK));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.MONTH));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.YEAR));
+                } else {
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.DAY));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.WEEK));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.MONTH));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.YEAR));
+                }
+
+                try {
+                    for (SolarEdgeCommand command : commands) {
+                        handler.getWebInterface().executeCommand(command);
+                        // we must not flood the API with requests, otherwise we will get an exception
+                        Thread.sleep(5000);
+                    }
+                } catch (RuntimeException e) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Caught Exception: {}", e.getMessage(), e);
+                    } else {
+                        logger.warn("Caught Exception: {}", e.getMessage());
```

Comment:
Is it guaranteed that we only get here in case of a software bug situation? If so, you should add the stacktrace as well.

Reply:
Most likely it is caused by an unexpected API response either a network error or an unexpected data format which is not handled correctly.
In case debug is enable we get a stacktrace. In case debug is disabled we only get the message.

Reply:
>  either a network error

Like not able to access the web server? In that case, you must not do any WARN logging, but set the Thing status to OFFLINE with an according message.

>  In case debug is disabled we only get the message.

What is the user supposed to do with this information? He will most likely report a bug and for this to be processed, the stacktrace will be very helpful. Asking the average user to enable debugging first will cause you/us unnecessary additional work.

Reply:
I removed the exception handling completely.